backends/keymapper/keymap.cpp

```diff
@@ -124,74 +124,108 @@ const Action *Keymap::findAction(const char *id) const {
 	return nullptr;
 }
 
-Keymap::ActionArray Keymap::getMappedActions(const Event &event) const {
+Keymap::KeymapMatch Keymap::getMappedActions(const Event &event, ActionArray &actions) const {
 	switch (event.type) {
 	case EVENT_KEYDOWN:
 	case EVENT_KEYUP: {
 		KeyState normalizedKeystate = KeyboardHardwareInputSet::normalizeKeyState(event.kbd);
 		HardwareInput hardwareInput = HardwareInput::createKeyboard("", normalizedKeystate, "");
-		return _hwActionMap[hardwareInput];
+		actions.push_back(_hwActionMap[hardwareInput]);
+		if (!actions.empty()) {
+			return kKeymapMatchExact;
+		}
+
+		if (normalizedKeystate.flags & KBD_NON_STICKY) {
+			// If no matching actions and non-sticky keyboard modifiers are down,
+			// check again for matches without the exact keyboard modifiers
+			for (HardwareActionMap::const_iterator itInput = _hwActionMap.begin(); itInput != _hwActionMap.end(); ++itInput) {
+				if (itInput->_key.type == kHardwareInputTypeKeyboard && itInput->_key.key.keycode == normalizedKeystate.keycode) {
+					int flags = itInput->_key.key.flags;
+					if (flags & KBD_NON_STICKY && (flags & normalizedKeystate.keycode) == flags) {
```

Comment:
Doing a bitwise and between the flags and the keycode does not seem correct.

Reply:
Yeah, that does seem wrong. Good eye.